core/app/models/spree/order/checkout.rb

```diff
@@ -229,11 +229,13 @@ def update_from_params(params, permitted_params, request_env = {})
             success = false
             @updating_params = params
             run_callbacks :updating_from_params do
+              existing_card_id = @updating_params[:order] ? @updating_params[:order].delete(:existing_card) : nil
+
               attributes = @updating_params[:order] ? @updating_params[:order].permit(permitted_params).delete_if { |k,v| v.nil? } : {}
 
               # Set existing card after setting permitted parameters because
               # rails would slice parameters containg ruby objects, apparently
-              existing_card_id = @updating_params[:order] ? @updating_params[:order][:existing_card] : nil
+             #  existing_card_id = @updating_params[:order] ? @updating_params[:order][:existing_card] : nil
```

Comment:
Could you please remove this line altogether now and move the comments above where you've moved this line to.